build.yaml

```diff
@@ -266,7 +271,19 @@ common_cli:
     # defintion, and feed it to the job configuration as the default value for
     # the 'MODULE' parameter.
     # This is how we tell trial which flocker module to call.
-    coverage run \$venv/bin/trial --reporter=subunit \$MODULE 2>&1 | parse_logs tee trial.log
+    coverage run \$venv/bin/trial \
+      --reporter=subunit \$MODULE 2>&1 | parse_logs tee trial.log
+
+  run_trial_with_coverage_as_root: &run_trial_with_coverage_as_root |
+    # The jobs.groovy.j2 file produces jobs that contain a parameterized job
+    # type. These type of jobs always require a parameter to be passed on in
+    # order for they to be executed.
+    # We grab the value from the 'with_modules:' dictionary in the yaml job
+    # defintion, and feed it to the job configuration as the default value for
+    # the 'MODULE' parameter.
+    # This is how we tell trial which flocker module to call.
+    sudo \$venv/bin/coverage run \$venv/bin/trial \
+      --reporter=subunit \$MODULE 2>&1 | parse_logs tee trial.log
```

Comment:
1. it is actually the same groovy job

on the build.yaml:
'run_trial_on_AWS_CentOS_7', 'run_trial_on_AWS_CentOS_7_as_root', 'run_trial_on_AWS_Ubuntu_Trusty', 'run_trial_on_AWS_Ubuntu_Trusty_as_root' as keys of job_type/run_trial

```
job_type:
  run_trial:
    # http://build.clusterhq.com/builders/flocker-centos-7
    run_trial_on_AWS_CentOS_7:
      on_nodes_with_labels: 'aws-centos-7-SELinux-T2Small'
      with_modules: *run_trial_modules
.
.
.
    run_trial_on_AWS_CentOS_7_as_root:
      on_nodes_with_labels: 'aws-centos-7-SELinux-T2Small'
      with_modules:
        - flocker.volume
      with_steps:
        - { type: 'shell', cli: *run_trial_cli_as_root }
```

and the type 'run_trial'  is rendered by:
```
{# apply config related to 'run_trial' jobs                                  #}
{%    if job_type == 'run_trial'                                            -%}
{%      for _module in job_values.with_modules                              -%}
{%        set _job_name = job_name + '_' + _module                          -%}
  job("${dashProject}/${dashBranchName}/{{ _job_name }}") {
    parameters {
{# we pass the 'MODULE' parameter as the flocker module to test with trial   #}
        textParam("MODULE", "{{ _module }}", "Module to test" )
        textParam("TRIGGERED_BRANCH", "${branchName}", "Branch that triggered this job" )
    }
{# limit execution to jenkins slaves with a particular label                 #}
    label("{{ job_values.on_nodes_with_labels }}")
{% set directory_to_delete = "/tmp/workspace/${dashProject}/${dashBranchName}/" + _job_name + "/_trial_temp" %}
    {{ wrappers(job_values, directory_to_delete) }}
    {{ scm("${git_url}", "${branchName}") }}
    {{ steps(job_values.with_steps) }}
    {{ publishers(job_values) }}
  }
{%      endfor                                                               %}
{%    endif         
```